### PR TITLE
Incremental Mongoose destruction #2559

### PR DESCRIPTION
**Pull request by iamleeg**

### CODE REVIEW

**Comment by iamleeg** [comment]
Firefox Cypress tests seem flaky: I get different failures locally and in different runs on CI.

**Comment by iamleeg** [comment]
Upgrading to latest Cypress causes different failures in `cypress-run-ff`, at least locally :(.

File: verification/curator-service/api/src/controllers/auth.ts [hunk 1]
```diff
@@ -442,20 +442,22 @@ export class AuthController {
                     }
 
                     // Check if there is a token in DB already for this user
-                    const token = await Token.findOne({ userId: user._id });
+                    const query = { userId: user._id };
+                    const token = await tokens().findOne(query);
                     if (token) {
-                        await token.deleteOne();
+                        await token().deleteOne(query);
```
Comment (abhidg) [inline verification/curator-service/api/src/controllers/auth.ts:448]: tokens()

**Comment by iamleeg** [comment]
What I'm seeing is that random tests fail with "expected to find content [whatever] but never did", when the browser screenshot shows the correct UI but not updated to reflect the actions that have been taken in the test (e.g. logged in on the line list, but not showing the filter UI when the test has applied a filter and wants to see the results). So seems like a timing problem but I don't see what has changed to cause it, unless there was recently a new FF version.

**Review by jim-sheldon — APPROVED** [review]

**Pull request merged**